Reorganize and edit the section "Restrict cross namespace resource associations" (#3430)
elastic/cloud-on-k8s: docs/operating-eck/restrict-cross-namespace-associations.asciidoc

****
endif::[]
[id="{p}-{page_id}"]
= Restrict cross-namespace resource associations

This section describes how to restrict associations that can be created between resources managed by ECK.

> When using the `elasticsearchRef` field to establish a trust relationship between Elasticsearch, Kibana, or APM Server resources, by default the association is allowed as long as both resources are deployed to namespaces managed by that particular ECK instance. The association is successful even if the user does not have access to one of the namespaces, or has only access to Kibana resources and not Elasticsearch resources.

Reviewer: ```suggestion
When using the `elasticsearchRef` field to establish a connection to Elasticsearch from Kibana, APM Server, Enterprise Search, or Beats resources, by default the association is allowed as long as both resources are deployed to namespaces managed by that particular ECK instance. The association will succeed even if the user creating the association does not have access to one of the namespaces or the Elasticsearch resource.
```

> The restriction of cross-namespace associations is disabled by default. Once enabled, it only enforces an access control for resources deployed across two different namespaces. You can still create associations between resources deployed in a same namespace.

Reviewer: ```suggestion
The enforcement of access control rules for cross-namespace associations is disabled by default. Once enabled, it only enforces access control for resources deployed across two different namespaces. Associations between resources deployed in the same namespace are not affected.
```

> Associations are allowed as long as the `ServiceAccount` specified in the associated resource can execute HTTP `GET` requests against the referenced Elasticsearch object.

Reviewer: ```suggestion
Associations are allowed as long as the `ServiceAccount` used by the associated resource can execute HTTP `GET` requests against the referenced Elasticsearch object.
```

  serviceAccountName: associated-resource-sa
----

> In the above example, `associated-resource` can be of any `Kind` that requires an association to be created, for example `Kibana` or the `APMServer`.

Reviewer: ```suggestion
In the above example, `associated-resource` can be of any `Kind` that requires an association to be created, for example `Kibana` or `ApmServer`.
```

You can find https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/associations-rbac/apm_es_kibana_rbac.yaml[a complete example in the ECK GitHub repository].

> NOTE: If the `serviceAccountName` is not set, ECK uses the `ServiceAccount` called `default`.

Reviewer: ```suggestion
NOTE: If the `serviceAccountName` is not set, ECK uses the default service account assigned to the pod by the link:https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-admission-controller[Service Account Admission Controller].
```
